packages/babel-plugin-remove-graphql-queries/src/index.ts

```diff
@@ -169,34 +249,38 @@ function getGraphQLTag(path) {
   }
 }
 
-function isUseStaticQuery(path) {
+function isUseStaticQuery(path: NodePath<CallExpression>): boolean {
   return (
     (path.node.callee.type === `MemberExpression` &&
       path.node.callee.property.name === `useStaticQuery` &&
-      path.get(`callee`).get(`object`).referencesImport(`gatsby`)) ||
-    (path.node.callee.name === `useStaticQuery` &&
-      path.get(`callee`).referencesImport(`gatsby`))
+      (path.get(`callee`).get(`object`) as NodePath).referencesImport(
+        `gatsby`,
+        ``
+      )) ||
+    ((path.node.callee as Identifier).name === `useStaticQuery` &&
+      path.get(`callee`).referencesImport(`gatsby`, ``))
```

Comment:
As an aside, I think the `callee` could be `foo[useStaticQuery]` and this check would pass because it should check `callee.computed` too, but that's a different matter :p